sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsScaleMonitor.cs

```diff
@@ -89,8 +89,9 @@ public async Task<EventHubsTriggerMetrics> GetMetricsAsync()
 
             try
             {
-                checkpoints = await Task.WhenAll(checkpointTasks)
-                    .ConfigureAwait(false);
+                checkpoints = (await Task.WhenAll(checkpointTasks).ConfigureAwait(false))
+                    .Where(c => c != null)
+                    .ToArray();
```

Comment:
Should we avoid allocating and just add the condition into the SingleOrDefault clause?